src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -223,6 +230,11 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     }
                     );
 
+                    if (clients.Count() < totalClientCount && _requestTracingEnabled && _requestTracingOptions != null)
```

Comment:
Comparing the client count can't tell failover happen or not, it just tell some client's backoff time is not due.

Reply:
https://github.com/Azure/AppConfiguration-DotnetProvider/pull/584#discussion_r1690967802

> Let's consider this concrete scenario, customer passes an endpoint (static) and we discover a replica endpoint (dynamic)
> 
> At a time point, the static one starts to fail, failover to the dynamic one, current code can handle the dynamic client as a failover, add failover tag in correlation context.
> The static client keeps failing and refresh trigger before its backoff time due, I think the dynamic client still need to have the failover tag in correlation context, but the current code will treat it as not a failover

For this scenario, if there is any client's backoff time is not due, it means failover happened before.

Reply:
I planned to move this PR on preview branch where we have load balancing implementated.